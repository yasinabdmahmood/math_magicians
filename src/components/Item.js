import React from 'react';
import PropTypes from 'prop-types';
// eslint-disable-next-line react/prefer-stateless-function
class Item extends React.Component {
  constructor(props) {
    super(props);
    this.handleObjectChange = this.handleObjectChange.bind(this);
  }

  handleObjectChange() {
    const { label, handleObjectChange } = this.props;
    handleObjectChange(label);
  }

  render() {
    const { label, cl } = this.props;
    return (

      <button
        className={`item   ${cl}`}
        type="submit"
        onClick={this.handleObjectChange}
      >
        {label}

      </button>

    );
  }
}
Item.propTypes = {
  label: PropTypes.string.isRequired,
  cl: PropTypes.string.isRequired,
  handleObjectChange: PropTypes.func.isRequired,

};

export default Item;
