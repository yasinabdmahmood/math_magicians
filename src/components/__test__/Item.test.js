import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Item from '../Item';

const mockFunction = jest.fn;
it('Check if Item component has changed', () => {
  const tree = renderer.create(
    <Item label="AC" cl="light-gray" handleObjectChange={mockFunction} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('check the input props and its effect on the component', () => {
  it('the button should have class item', () => {
    render(
      <Item label="AC" cl="light-gray" handleObjectChange={mockFunction} />,
    );
    const button = screen.getByRole('button');
    expect(button).toHaveClass('item');
  });
  it('the button should have class light-gray', () => {
    render(
      <Item label="AC" cl="light-gray" handleObjectChange={mockFunction} />,
    );
    const button = screen.getByRole('button');
    expect(button).toHaveClass('light-gray');
  });
  it('the button should have text  AC ', () => {
    render(
      <Item label="AC" cl="light-gray" handleObjectChange={mockFunction} />,
    );
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('AC');
  });
});
