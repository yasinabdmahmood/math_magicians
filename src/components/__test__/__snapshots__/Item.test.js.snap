// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Item component has changed 1`] = `
<button
  className="item   light-gray"
  onClick={[Function]}
  type="submit"
>
  AC
</button>
`;
